### ISSUE TITLE
Menu cart reloads when a product is deleted

### ISSUE THREAD

**Issue by SeriouZGame** [elementor/elementor#9531]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description

When a product is put in the cart and you delete it in the menu cart, the page reloads and the cart closes, it's not really user friendly... You say menu cart comes with ajax out of the box, but the page reloads every time you delete a product... how can I solve this?

## Steps to reproduce

Put a product in the menu cart and then delete it.

## Isolating the problem

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>
```
<!-- For bug reports, let us know about your system environment: Copy and paste the system info report from Elementor => System info, and paste it here or in http://pastebin.com/ -->
```
</details>

**Comment by Stilian1** [comment]
I have the same issue.

When I delete an item from inside the Menu Cart, the page refreshes, and a WooCommerce notice shows on the page saying "[item] removed".

This has been reported again more that a year ago here: [https://github.com/elementor/elementor/issues/5634](url). And @bainternet said that AJAX is supported. Does that mean that the page shouldn't be refreshed when the cart is updated via the Elementor Menu Cart or not? If so, then why it does? Is this a conflict with my and other users' WordPress environment? @bainternet, please clarify.

**Comment by Creativenauts** [comment]
I just tested this on my site and can confirm that the delete button on menu cart just refreshes the page and it doesn't remove the item from the cart.

**Comment by amundse** [comment]
I have the same issue. I don't think it's a bug but rather a question of functionality. From a user point of view it would be much better if only the cart is updated, without the page refresh. I hope this can be solved.

**Comment by flagu** [comment]
Please add class="remove remove_from_cart_button" to the mini-cart.php file in line 54 to make the Ajax delete function of the shopping cart widget in WooCommerce work again.

**Comment by IlPizza79** [comment]
Thank you @flagu, that's what I need :-)
Elementor staff, please add this small fix in the next release!

**Comment by IlPizza79** [comment]
Elementor staff, please add class="remove remove_from_cart_button" to the mini-cart.php.
It's just a small and easy fix, please :-(

**Comment by flagu** [comment]
Instead of:
`echo apply_filters( 'woocommerce_cart_item_remove_link', sprintf(
					'<a href="%s" aria-label="%s" data-product_id="%s" data-cart_item_key="%s" data-product_sku="%s"></a>',`

it should be:
`echo apply_filters( 'woocommerce_cart_item_remove_link', sprintf(
					'<a href="%s" **class="remove remove_from_cart_button"** aria-label="%s" data-product_id="%s" data-cart_item_key="%s" data-product_sku="%s"></a>',`

**Comment by flagu** [comment]
> Please add class="remove remove_from_cart_button" to the mini-cart.php file in line 54 to make the Ajax delete function of the shopping cart widget in WooCommerce work again.

Instead of:
echo apply_filters( 'woocommerce_cart_item_remove_link', sprintf( '<a href="%s" aria-label="%s" data-product_id="%s" data-cart_item_key="%s" data-product_sku="%s"></a>',

it should be:
echo apply_filters( 'woocommerce_cart_item_remove_link', sprintf( '<a href="%s" class="remove remove_from_cart_button" aria-label="%s" data-product_id="%s" data-cart_item_key="%s" data-product_sku="%s"></a>',

**Comment by anakhha** [comment]
Unfortunately I can also confirm that. The error is still present up to Elementor version 3.2. @elementor Team, is there a short-term solution?

**Comment by stannvanheusden** [comment]
If you are looking for a good short-term solution, put this code to your functions.php: 

`function elefix_menucart(){ ?>
<script type="text/javascript" defer="defer">
setInterval(function(){ 
    jQuery('.elementor-menu-cart__product-remove a').addClass('remove remove_from_cart_button');
}, 3000);
</script>
<?php }
add_action('wp_footer', 'elefix_menucart');`

What this code does: it adds a jQuery snippet to your website that will add the classes to the remove button. It runs every 3 seconds because a client can add new items while on the page. 

Please note: This is not a permanent fix since there are better ways to do it. Elementor should fix it in an update, but until then I think this is the best way.

**Comment by shilo-ey** [comment]
This feature has been implemented in Elementor Pro 3.4.0

Feel free to update

Thanks!
